---
description: 了解详细信息： "Throw" 语句在 "Catch" 语句外或 "Finally" 语句内不能省略操作数
title: “Throw”语句在“Catch”语句外或“Finally”语句内不能省略操作数
ms.date: 07/20/2015
f1_keywords:
- vbc30666
- bc30666
helpviewer_keywords:
- BC30666
ms.assetid: a208a6ea-0e36-4bf1-8984-4de1a0e38a2a
ms.openlocfilehash: 8ab76ae9e5727388ddb77ec38f1b4e89b6a0e5d6
ms.sourcegitcommit: 10e719780594efc781b15295e499c66f316068b8
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 02/14/2021
ms.locfileid: "100460834"
---
# <a name="throw-statement-cannot-omit-operand-outside-a-catch-statement-or-inside-a-finally-statement"></a><span data-ttu-id="931d4-103">“Throw”语句在“Catch”语句外或“Finally”语句内不能省略操作数</span><span class="sxs-lookup"><span data-stu-id="931d4-103">'Throw' statement cannot omit operand outside a 'Catch' statement or inside a 'Finally' statement</span></span>

<span data-ttu-id="931d4-104">`Throw` 语句外的 `Catch` 语句必须提供异常对象的名称。</span><span class="sxs-lookup"><span data-stu-id="931d4-104">`Throw` statements outside of `Catch` statement must supply the name of an exception object.</span></span>  
  
 <span data-ttu-id="931d4-105">**错误 ID：** BC30666</span><span class="sxs-lookup"><span data-stu-id="931d4-105">**Error ID:** BC30666</span></span>  
  
## <a name="to-correct-this-error"></a><span data-ttu-id="931d4-106">更正此错误</span><span class="sxs-lookup"><span data-stu-id="931d4-106">To correct this error</span></span>  
  
1. <span data-ttu-id="931d4-107">指定派生自 `System.Exception`的异常对象的名称。</span><span class="sxs-lookup"><span data-stu-id="931d4-107">Specify the name of an exception object derived from `System.Exception`.</span></span>  
  
2. <span data-ttu-id="931d4-108">重构代码，以便 `Throw` 语句位于 `Catch` 块中。</span><span class="sxs-lookup"><span data-stu-id="931d4-108">Restructure your code so that the `Throw` statement is inside a `Catch` block.</span></span>  
  
## <a name="see-also"></a><span data-ttu-id="931d4-109">请参阅</span><span class="sxs-lookup"><span data-stu-id="931d4-109">See also</span></span>

- [<span data-ttu-id="931d4-110">Throw 语句</span><span class="sxs-lookup"><span data-stu-id="931d4-110">Throw Statement</span></span>](../language-reference/statements/throw-statement.md)
- [<span data-ttu-id="931d4-111">Try...Catch...Finally 语句</span><span class="sxs-lookup"><span data-stu-id="931d4-111">Try...Catch...Finally Statement</span></span>](../language-reference/statements/try-catch-finally-statement.md)
- <xref:System.Exception?displayProperty=nameWithType>
- [<span data-ttu-id="931d4-112">在 .NET 中处理和引发异常</span><span class="sxs-lookup"><span data-stu-id="931d4-112">Handling and throwing exceptions in .NET</span></span>](../../standard/exceptions/index.md)
